components/infobox/extensions/commons/infobox_extension_game_appearances.lua

```diff
@@ -0,0 +1,45 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Infobox/Extension/GameApperances
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Game = require('Module:Game')
+local Table = require('Module:Table')
+
+local DEFAULT_MAX_NUMBER_OF_PLAYERS_IN_PLACEMENT = 10
+
+local Appearances = {}
+
+----Provide a list of games compete by a player
+---@param args {player: string?, numberOfPlayersInPlacement}?
+---@return string[]?
+function Appearances.player(args)
+	if not args or not args.player then return end
+	local numberOfPlayersInPlacement = args.numberOfPlayersInPlacement or DEFAULT_MAX_NUMBER_OF_PLAYERS_IN_PLACEMENT
+
+	local conditions = Array.map(Array.range(1, numberOfPlayersInPlacement), function(index)
+		return '[[opponentplayers_p' .. index .. '::' .. args.player .. ']]'
+	end)
+	table.insert(conditions, '[[opponentname::' .. args.player .. ']]')
+
+	local queriedGames = Array.map(mw.ext.LiquipediaDB.lpdb('placement', {
+		conditions = table.concat(conditions, ' OR '),
+		query = 'game',
+		groupby = 'game asc',
+		limit = 1000,
+	}), function(item) return item.game end)
+
+	local orderedGames = Array.filter(Game.listGames{ordered = true}, function(game)
+		return Table.includes(queriedGames, game)
```

Comment:
```suggestion
	local queriedGames = Table.map(mw.ext.LiquipediaDB.lpdb('placement', {
		conditions = table.concat(conditions, ' OR '),
		query = 'game',
		groupby = 'game asc',
		limit = 1000,
	}), function(_, item) return item.game, true end)

	local orderedGames = Array.filter(Game.listGames{ordered = true}, function(game)
		return queriedGames[game]
```

Would make it a bit more performant

Reply:
```suggestion
	local queriedGames = Table.map(mw.ext.LiquipediaDB.lpdb('placement', {
		conditions = table.concat(conditions, ' OR '),
		query = 'game',
		groupby = 'game asc',
		limit = 1000,
	}), function(_, item) return Game.toIdentifier{game = item.game, useDefault = false}, true end)

	local orderedGames = Array.filter(Game.listGames{ordered = true}, function(game)
		return queriedGames[game]
```

if you want to make sure as per the comment above

Reply:
ok, now theres a new issue that I discovered 

(e.g. <https://liquipedia.net/simracing/CGS/2008>)
- The game doesnt generate into the list if the player competes in only events that uses Team Prize Pool (and never compete one under that game that uses a Solo Prize Pool)

Reply:
The problem is not that, the problem is that there's no matches and you're querying on matches

Reply:
Underscore vs non-underscore in storage it seems

Reply:
any possible way to have the extension catch the underscores (redirection?) 
(the game entry in CGS 2007 does matched the entry input in Info module tho) 
![image](https://github.com/Liquipedia/Lua-Modules/assets/88981446/eddca7a6-64da-49ac-b2e3-22318cd6eefd)

Reply:
pgr3 is in Info

Reply:
> pgr3 is in Info

yeah I'm aware of that, Like I thought its a mismatched stored entry between Info vs Placement data but its not
is there a way to fix this, underscore vs non underscores seems like outside the extension?

Reply:
Query for both